src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/UserService.kt

```diff
@@ -1,20 +1,53 @@
 package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service
 
+import org.springframework.security.access.AccessDeniedException
 import org.springframework.stereotype.Service
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.client.manageusers.ManageUsersClient
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.client.manageusers.model.UserDetailsDto.AuthSource
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.common.isEmail
-import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.entity.ContactDetails
 
 @Service
 class UserService(private val manageUsersClient: ManageUsersClient) {
 
-  fun getContactDetails(username: String): ContactDetails? {
-    val mayBeUserDetails = manageUsersClient.getUsersDetails(username) ?: return null
+  companion object {
+    private enum class ServiceName {
+      BOOK_A_VIDEO_LINK_SERVICE,
+    }
+
+    fun getServiceAsUser() = User(
+      username = ServiceName.BOOK_A_VIDEO_LINK_SERVICE.name,
+      userType = UserType.SERVICE,
+      name = ServiceName.BOOK_A_VIDEO_LINK_SERVICE.name,
+    )
+
+    fun getClientAsUser(clientId: String) = User(
+      username = clientId,
+      userType = UserType.SERVICE,
+      name = clientId,
+    )
+  }
 
-    return if (username.isEmail()) {
-      ContactDetails(name = mayBeUserDetails.name, email = username)
-    } else {
-      return ContactDetails(name = mayBeUserDetails.name, email = manageUsersClient.getUsersEmail(username)?.email)
+  fun getUser(username: String): User? {
+    val userDetails = manageUsersClient.getUsersDetails(username) ?: return null
+    val userType = when (userDetails.authSource) {
+      AuthSource.nomis -> UserType.PRISON
+      AuthSource.auth -> UserType.EXTERNAL
+      else -> throw AccessDeniedException("Users with auth source ${userDetails.authSource} are not supported by this service")
     }
+
+    return User(
+      username = username,
+      userType = userType,
+      name = userDetails.name,
+      email = if (username.isEmail()) username.lowercase() else manageUsersClient.getUsersEmail(username)?.email,
```

Comment:
Not sure this really matters but for consistency I would lowercase the email on the call to the client as well ...

`manageUsersClient.getUsersEmail(username)?.email?.lowercase()`